opentelemetry-appender-log/src/lib.rs

```diff
@@ -70,13 +74,569 @@ const fn severity_of_level(level: Level) -> Severity {
     }
 }
 
-#[cfg(all(test, feature = "testing", feature = "logs"))]
+fn log_attributes(kvs: impl log::kv::Source) -> Vec<(Key, AnyValue)> {
+    struct AttributeVisitor(Vec<(Key, AnyValue)>);
+
+    impl<'kvs> log::kv::VisitSource<'kvs> for AttributeVisitor {
+        fn visit_pair(
+            &mut self,
+            key: log::kv::Key<'kvs>,
+            value: log::kv::Value<'kvs>,
+        ) -> Result<(), log::kv::Error> {
+            let key = Key::from(String::from(key.as_str()));
```

Comment:
Does log's Key API provides way to check if the key is a borrowed reference, or an owned string. As otel::key internally stores string reference, so we can avoid the allocation if possible ?

Reply:
It doesn’t currently, but we’ve left the door open for it.